src/System.Collections/tests/Generic/Dictionary/Dictionary.Tests.cs

```diff
@@ -225,6 +225,19 @@ public void CopyTo_NonContiguousDictionary(int count)
                 Assert.Equal(array[i++], obj.Key);
         }
 
+        [Fact]
+        public void Remove_NonExistentEntries_DoesNotPreventEnumeration()
+        {
+            var dictionary = new Dictionary<string, string>();
+            dictionary.Add("a", "b");
+            dictionary.Add("c", "d");
+            foreach (var entry in dictionary)
+            {
+                if (dictionary.Remove(entry.Key + "Sibling"))
```

Comment:
I assume instead of "Sibling" it could just be `if (dictionary.Remove("nonexistent_key")`

Or is there some importance in it sorting between the two key values that I am missing? @benaadams

Reply:
Well... it could also be an actual key as long as the enumeration doesn't continue so; covering both:
```csharp
public void Remove_NonExistentEntries_DoesNotPreventEnumeration()
{
    var dictionary = new Dictionary<string, string>();
    dictionary.Add("a", "b");
    dictionary.Add("c", "d");
    foreach (var entry in dictionary)
    {
        if (dictionary.Remove(entry.Key + "Sibling"))
            break;
    }

    dictionary.Add("cSibling", "d");
    foreach (var entry in dictionary)
    {
        if (dictionary.Remove(entry.Key + "Sibling"))
            break;
    }
}
```

Reply:
Need two loops incase the enumeration returns `c` as the first item then doesn't continue passed `Remove`

Reply:
Still don't know what is meant by Sibling -- but fine, that works